firebase.go

```diff
@@ -105,6 +105,23 @@ func (fb *Firebase) Unauth() {
 	fb.params.Del(authParam)
 }
 
+// SetRef changes the current reference location whilst keeping the host url.
+func (fb *Firebase) SetRef(path string) {
+	parsedURL, err := _url.Parse(fb.url)
+	if err != nil {
+		panic(err)
+	}
+	if strings.Index(path, "/") == 0 {
```

Comment:
I would suggest doing something like

```go
path = strings.Trim(path, "/")
```

So that we can trim the leading and trailing slashes